dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/SimpleRotatingSoftQueue.java

```diff
@@ -74,27 +74,15 @@ private int getNext(int index) {
         return (++index == elements.size()) ? 0 : index;
     }
 
-    /**
-     *
-     * @return an atomic copy of the queue. The elements in the returned array
-     *         are in order from head to tail so it can be iterated normally.
-     *
-     */
     @Override
-    public <T> T[] toArray(T[] arr) {
-
-        if (arr == null) {
-            return null;
-        }
+    public Object[] toArray() {
```

Comment:
WsTraceRouterImpl:80 still uses toArray(T[] arr)
To avoid duplicate code, either switch any calls to toArray(T[] arr) to use toArray() or switch BufferManager to use toArray(T[] arr)

The return time should also be T[] instead of Object[] if you decide to keep toArray()